### ISSUE TITLE
[New Feature Request] Set PyTorch batch size appropriately

### ISSUE THREAD

**Issue by rpreen** [AI-SDC/AI-SDC#447]
Currently the [PyTorch wrapper](https://github.com/AI-SDC/SACRO-ML/blob/main/sacroml/attacks/model_pytorch.py) sets the batch size to 32, but it should instead use the same as the original training because it has a large influence on performance. 

I think the reason 32 was originally set was since the hardware used to train the model might differ from the hardware used to assess it - if it was trained on a machine with a giant GPU it may not run at all on a smaller one. A possible solution to that might be to use linear scaling of the learning rates proportional to the batch size as this seems to be used in practice to achieve roughly comparable nets (bigger batch size = bigger learning rate) - but if done then this should be shown clearly as a warning to the person running something like LiRA or perhaps presented as an option.

A complicating factor is that the `batch_size` parameter is not currently supplied in the `train_params` because it has been considered part of the dataset handler, for example:

https://github.com/AI-SDC/SACRO-ML/blob/a7ebc8e141556d62064839200fa9c504e48ef7d4/examples/pytorch/simple/dataset.py#L61-L70

and that function is used in the `Target` class when loading the dataset initially, but then undergoes the conversion from dataloader to numpy as explained in #411 

https://github.com/AI-SDC/SACRO-ML/blob/a7ebc8e141556d62064839200fa9c504e48ef7d4/sacroml/attacks/target.py#L191-L197

A possible solution might be if #411 was addressed and the train/test indices for the different shadow models were passed to the `get_dataloader()` function in the supplied dataset module - but that seems like a big change. 

**For now, if might be simpler to capture the `batch_size` explicitly as an optional parameter in the `Target` constructor (and update the examples)  and then use that in the `PytorchModel` class.**
